bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -629,7 +629,32 @@ public static boolean checkIsType(Type sourceType, Type targetType, List<TypePai
                 if (sourceTypeTag == TypeTags.FINITE_TYPE_TAG) {
                     return isFiniteTypeMatch((BFiniteType) sourceType, targetType);
                 }
-                return sourceTypeTag == TypeTags.BYTE_TAG || sourceTypeTag == TypeTags.INT_TAG;
+                return sourceTypeTag == TypeTags.BYTE_TAG || TypeTags.isIntegerTypeTag(sourceTypeTag);
+            case TypeTags.SIGNED16_INT_TAG:
+                if (sourceTypeTag == TypeTags.FINITE_TYPE_TAG) {
+                    return isFiniteTypeMatch((BFiniteType) sourceType, targetType);
+                }
+                return sourceTypeTag == TypeTags.BYTE_TAG || sourceTypeTag == TypeTags.UNSIGNED8_INT_TAG ||
+                        sourceTypeTag == TypeTags.SIGNED8_INT_TAG || sourceTypeTag == TypeTags.SIGNED16_INT_TAG;
+            case TypeTags.SIGNED32_INT_TAG:
+                if (sourceTypeTag == TypeTags.FINITE_TYPE_TAG) {
+                    return isFiniteTypeMatch((BFiniteType) sourceType, targetType);
+                }
```

Comment:
can we move this check before the switch statement if this is repeating for most of the cases ?

Reply:
Addressed with https://github.com/ballerina-platform/ballerina-lang/pull/31302/commits/49f13546fca88ec1d4bcfc36d1d6ba6b7de801ce